src/main/java/net/kyori/adventure/text/minimessage/transformation/Util.java

```diff
@@ -0,0 +1,17 @@
+package net.kyori.adventure.text.minimessage.transformation;
+
+import org.checkerframework.checker.nullness.qual.NonNull;
+
+public class Util {
+
+    public static @NonNull String stripOuterQuotes(final @NonNull String string) {
+        if(string.length() != 1 && startsAndEndsWithQuotes(string)) {
+            return string.substring(1, string.length() - 2);
+        }
+        return string;
+    }
+
+    public static boolean startsAndEndsWithQuotes(final @NonNull String string) {
+        return string.startsWith("'") && string.endsWith("'") || string.startsWith("\"") && string.endsWith("\"");
+    }
```

Comment:
This shouldn't be necessary as the parser does this already:

https://github.com/KyoriPowered/adventure-text-minimessage/blob/564ecffbe1b59b010c8b4ffff77a97cad8d17803/src/main/grammars/MiniParser.jj#L167-L173

Also `string.length() - 2` is incorrect.